smalltalksrc/Slang/TMethod.class.st

```diff
@@ -651,9 +658,17 @@ TMethod >> declarationAt: aVariableName ifPresent: presentBlock [
 ]
 
 { #category : #accessing }
-TMethod >> declarationAt: aVariableName  "<String>" put: aDeclaration [ "<String>" "^aDeclaration"
+TMethod >> declarationAt: aVariableName put: aDeclaration [
 
-	^declarations at: aVariableName put: aDeclaration
+	| isRelevantVariable |
+	isRelevantVariable := (aVariableName beginsWithAnyOf:
+		                       #( #self , #cascade, #toDoLimit )) not.
+
+	parseTree notNil and: [
```

Comment:
Had to add this validation
```st
parseTree notNil
```
Because some tests hadn't generated the `parseTree` maybe a possible improvement could be to modify those tests so the `parseTree` isn't `nil`

Reply:
@ivojawer do you want to do a pass on this PR or do you prefer to do a separate one?

Reply:
Please tell us if you're having troubles with this

Reply:
Sorry for leaving this comment hanging :sweat: 

I just pushed the changes we were talking about, what I had the hardest time with was finding which node could be used as the `parseTree`. I added an empty `TStatementListNode` as it was the only one that implemented the `locals` accessor but I'm still not 100% sure of it.

Reply:
it was the right choice!! thanks!